ntp/src/Ntp/Util.hs

```diff
@@ -244,27 +216,27 @@ sendPacket
     -> NtpPacket
     -> [Addresses]
     -> IO ()
-sendPacket tracer sock packet addrs = do
-    let bs = LBS.toStrict $ encode $ packet
-    traverse_
-        (\addr ->
-            (sendTo sock bs addr)
-                `catch` handleSendToException addr
-        )
-        addrs
+sendPacket tracer sock packet addrs
+    = traverse_ (\addr -> sendTo addr) addrs
   where
-    handleSendToException :: Addresses -> SendToException -> IO ()
-    handleSendToException addr NoMatchingSocket =
-        traceWith tracer $ NtpTraceSendPacketNoMatchingSocket (show addr) (show sock)
-    handleSendToException addr (SendToIOException addressFamily ioerr) = do
-        traceWith tracer $ NtpTraceSentToIOException (show addressFamily) ioerr
-        case (addr, addressFamily) of
-            -- try to send the packet to the other address in case the current
-            -- system does not support IPv4/6.
-            (These _ r, IPv6) -> do
-                traceWith tracer $ NtpTraceSentTryResend $ show $ runWithAddrFamily $ getFirst r
-                sendPacket tracer sock packet [That r]
-            (These l _, IPv4) -> do
-                traceWith tracer $ NtpTraceSentTryResend $ show $ runWithAddrFamily $ getFirst l
-                sendPacket tracer sock packet [This  l]
-            _                 -> traceWith tracer $ NtpTraceSentNotRetrying
+    bs :: ByteString
+    bs = LBS.toStrict $ encode $ packet
+
+    sendTo
+        :: Addresses
+        -- ^ addresses to send to
+        -> IO ()
+    sendTo addr = case pairThese sock addr of
+        Just s -> foldThese $ bimap fn fn s
```

Comment:
where does `foldThese` come from?  Wouldn't `foldMap` (or `bifoldMap` from `bifoldable`) just work (note `IO ()` is a monoid)

Reply:
Yes..this was just left unchanged with the `foldThese` from within that module.